include/sharg/concept.hpp

```diff
@@ -0,0 +1,52 @@
+// -----------------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/sharg-parser/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \brief Provides helper concepts.
+ */
+
+#pragma once
+
+#include <sharg/auxiliary.hpp>
+
+namespace sharg
+{
+
+/*!\concept sharg::istreamable
+ * \ingroup argument_parser
+ * \brief Concept for types that can be parsed from a std::istream via the stream operator.
+ * \tparam value_type The type to check whether it's stremable via std::istream.
+ *
+ * ### Requirements
+ *
+ * `std::istream` must support the (un)formatted input function (`operator>>`) for an l-value of a given `value_type`.
+ *
+ * todo: link to how-to page how to make your type istreamable
+ */
+template <typename value_type>
+concept istreamable = requires (std::istream & is, value_type & val)
+{
+    { is >> val };
```

Comment:
not sure, but should we also check for return type?

```suggestion
    { is >> val } -> std::is_same<std::istream&>;
```